moveit_simple/src/moveit_simple.cpp

```diff
@@ -443,66 +448,139 @@ bool Robot::jointInterpolation(const std::unique_ptr<TrajectoryPoint> & traj_poi
+                        << (num_steps+1) << "points for " << traj_point->name()
+                        << ". Clearing points added in unsuccessful attempt");
+      for (std::size_t j=0; j<points_added; ++j)
+      {
+        points.pop_back();
+      }
+      return false;
+    }
+  }
   return true;
 }
 
+
 bool Robot::cartesianInterpolation(const std::unique_ptr<TrajectoryPoint> & traj_point,
            std::vector<trajectory_msgs::JointTrajectoryPoint> & points,
            unsigned int num_steps)
 {
   trajectory_msgs::JointTrajectoryPoint prev_point_info = points.back();
-  // Covert the previous point stored in points to Cartesian Trajectory Point
+  // Convert the previous point stored in points to Cartesian Trajectory Point
   std::unique_ptr<TrajectoryPoint>prev_point =  std::unique_ptr<TrajectoryPoint>
 (new JointTrajectoryPoint(prev_point_info.positions, prev_point_info.time_from_start.toSec(), ""));
   std::unique_ptr<CartTrajectoryPoint>prev_traj_point =
   prev_point->toCartTrajPoint(*this);
+  std::unique_ptr<CartTrajectoryPoint> target_point;
   if (prev_traj_point)
   {
-    std::unique_ptr<CartTrajectoryPoint> target_point = traj_point->toCartTrajPoint(*this);
-    if (target_point)
+    target_point = traj_point->toCartTrajPoint(*this);
+    if (!target_point)
     {
-      unsigned int points_added = 0;
-      for (std::size_t i=1; i<=num_steps+1; ++i)
-      {
-        double t = (double)i / (double)(num_steps+1);
-        std::unique_ptr<TrajectoryPoint> new_point;
-        interpolate(prev_traj_point,target_point,t,new_point);
-
-        // New cartesian point can be added by joint interpolation
-        if(jointInterpolation(new_point, points))
-        {
-          points_added++;
-          ROS_INFO_STREAM( points_added << " points among " << (num_steps+1) << " added successfullyfor " << traj_point->name());
-        }else{
-          ROS_ERROR_STREAM("Conversion to joint trajectory failed at " << i << " among "
-                            << (num_steps+1) << "points for " << traj_point->name());
-          return false;
-        }
-      }
-        return true;
-    }else{
       ROS_ERROR_STREAM("Failed to find FK for " << traj_point->name());
       return false;
     }
+  }else{
     ROS_ERROR_STREAM("Failed to find FK for already added point. This is unexpected." );
     return false;
   }
+  unsigned int points_added = 0;
+  for (std::size_t i=1; i<=num_steps+1; ++i)
+  {
+    double t = (double)i / (double)(num_steps+1);
+    std::unique_ptr<CartTrajectoryPoint> new_point_cart;
+    interpolate(prev_traj_point,target_point,t,new_point_cart);
+
+    std::unique_ptr<TrajectoryPoint> new_point =std::unique_ptr<TrajectoryPoint>
+               ( new CartTrajectoryPoint(new_point_cart->pose(), new_point_cart->time(), ""));
+
+    // Add new point at the end of Joint Trajectory (named points)
+    if(jointInterpolation(new_point, points,(unsigned int) 0))
```

Comment:
It adds `num_steps` interpolated points + traj_point from Trajectory (added with addTrajectoryPoint)